contents/teams/feature-flags/objectives.mdx

```diff
@@ -1,53 +1,55 @@
-  - Create seamless workflow from flag creation to code implementation
-  - Implement AI-assisted troubleshooting for flag implementation issues
-
-- Developer Experience
-  - Measure time-to-implementation improvements
-  - Collect developer feedback on integration experience
-  - Iterate based on usage patterns and pain points
\ No newline at end of file
+## Q3 2025 Objectives
+
+### Objective 1: Behavioral Cohort Targeting
+
+Motivation: Enable dynamic behavioral targeting across PostHog – let customers target users based on what they've done (or haven't done) recently. In other words, enable customers to target feature flags on [behavioral cohorts](/docs/data/cohorts).
+
+This is a BHAG shared across multiple teams ([Messaging](/teams/messaging), [Feature Flags](/teams/feature-flags), [Ingestion](/teams/ingestion), and [Surveys](/teams/surveys)) who all need behavioral targeting to enhance their products. We'll build a platform that integrates with our ingestion pipeline to pre-process event data for targeting, then integrate this service into existing product targeting functionality.
+
+We'll deliver this in 3 stages with Stage 1 well-defined and Stages 2-3 evolving based on how Stage 1 goes.
+
+#### Stage 1: Proof-of-concept
+
+Build a proof of concept behavioral cohort data ingestion service as the foundation. Owned by <TeamMember name ="Ben White" photo /> (CDP/Messaging) and <TeamMember name="Dylan Martin" photo /> (Feature Flags).
+
+What we will ship:
+* **Support behavioral cohort filter as Hog bytecode** – Make cohort filters representable as Hog code for our CDP pipeline <TeamMember name="Dylan Martin" photo />
+* **Build system for loading/matching/saving targeting results** – Create a system that writes matches to a low-latency datastore (not Clickhouse) that other applications can query <TeamMember name="Ben White" photo />
+* **Validate with ingestion team** – Ensure the POC doesn't negatively impact ingestion latency or scale (TBD owner from [Ingestion](/teams/ingestion))
+
+#### Stage 2: Make it scale
+
+Implement the scaling solution based on POC learnings and ingestion team collaboration. Specific work TBD based on Stage 1 results.
+
+#### Stage 3: Integrate across PostHog
+
+Add behavioral cohort targeting to existing products using our new low-latency datastore.
+
+What we will ship:
+* **Behavioral Cohort Targeting for Feature Flags** – Add behavioral cohorts to existing feature flag targeting <TeamMember name ="Dylan Martin" photo />
+* **Behavioral Cohort Targeting for Surveys** – Add behavioral cohorts to existing survey targeting (owner TBD)
+* **Behavioral Cohort Targeting for Messaging** – Enable behavioral targeting in messaging (owner TBD)
+
+Finally, partner with 5-10 key customers to validate use cases, gather feedback, and measure impact on conversion and engagement <TeamMember name="Dylan Martin" photo /> and <TeamMember name="Annika Schmid" photo />
```

Comment:
Depending on how stage 1 goes, maybe we can partner with them earlier. Their feedback might inform how stage 2 and 3 pan out.